packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/QuestionCard.tsx

```diff
@@ -59,141 +63,184 @@ const QuestionCard: React.FC<QuestionCardProps> = ({
+                  overlayStyle={{ maxWidth: '60em' }}
+                >
+                  <div
+                    style={
+                      {
+                        // shorten question text dynamically
+                        display: '-webkit-box',
+                        WebkitLineClamp: 1,
+                        WebkitBoxOrient: 'vertical',
+                        textOverflow: 'ellipsis',
+                        overflow: 'hidden',
+                        maxWidth: '55em',
+                      } as React.CSSProperties
+                    }
+                  >
+                    {`${question.text}`}
+                  </div>
+                </Tooltip>
+              </>
+            )}
+            {isStaff && (
+              <div
+                className={cn(
+                  'itali mr-1 mt-0.5 inline-block min-w-[120px] text-sm',
+                  question.status === LimboQuestionStatus.ReQueueing
+                    ? 'text-gray-400'
+                    : 'text-gray-600',
+                )}
               >
-                {question.text}
+                {queueType === 'hybrid' && (
+                  <i>{`[${question.location ?? 'Unselected'}] `}</i>
+                )}
+                {question.creator.name}
               </div>
-            </Tooltip>
-          )}
-          {isStaff && (
-            <div className="mr-1 mt-0.5 inline-block min-w-[120px] text-sm italic text-gray-600">
-              {question.creator.name}
-            </div>
-          )}
+            )}
 
-          {question.questionTypes?.map((questionType, index) => (
-            <QuestionTagElement
-              key={index}
-              tagName={questionType.name}
-              tagColor={questionType.color}
-            />
-          ))}
-        </Col>
-        {isBeingHelped && !isStaff && question.helpedAt && (
-          <Col flex="0 0 3rem">
-            <div className="text-sm font-medium text-green-700">
-              {servedTime}
-            </div>
+            {question.questionTypes?.map((questionType, index) => (
+              <QuestionTagElement
+                key={index}
+                tagName={questionType.name}
+                tagColor={
+                  question.status !== LimboQuestionStatus.ReQueueing
+                    ? questionType.color
+                    : '#f0f0f0'
+                }
+              />
+            ))}
           </Col>
-        )}
-        <Col flex="0 0 3rem">
-          <div className="text-sm text-gray-600">{getWaitTime(question)}</div>
-          {isStaff && isBeingHelped && (
-            <div className="text-sm font-medium text-green-700">
-              {servedTime}
-            </div>
+          {question.status === LimboQuestionStatus.ReQueueing && (
+            <Col flex="1 1">
+              <div className="text-md h-full italic text-gray-600">
+                In the process of requeuing...
+              </div>
+            </Col>
+          )}
```

Comment:
This *cannot* be put here, as it will cause question text to be truncated early and also cause a lot of ugly wrapping.
![image](https://github.com/user-attachments/assets/d7512dbb-7b90-4000-894a-b081d6781f56)
![image](https://github.com/user-attachments/assets/684dbd23-83c9-4675-a4ca-8bd573909db5)

For a solution I have two ideas:
1. Floating text, similar to the "Currently Being Helped". I like this since this idea is already being used for showing question status, but I also don't like it since it gives these requeuing questions more attention than I would like (ideally, requeing questions get less emphasis compared to other questions, not more).
2. *Remove the TA buttons* since they are all disabled anyway, and put the text there instead

I think in option 2 may be better for TA on desktop, though it may look a little weird for TA on mobile. Though, the lack of buttons may also draw more attention to it as well.

For the student's side, I think the best option may be just to put it as the floating text, as much as it's not an ideal solution. This is simply because there is so little space on mobile.
![image](https://github.com/user-attachments/assets/e0ca2cf7-89a1-4941-a17c-4907d88010b1)

For consistency, you could just do floating text for both TA and student side. (This might be the best option overall). This may require some more discussion on Tuesday

Also, I would change the text to just say "Requeing..." or possibly "Not Ready" (the latter may be more universally understandable)

Reply:
Yea, when I added this for feedback, I didn't like it either but just didn't think of a good alternative. I agree that your latter suggestion makes more sense. Thanks for the idea! I'll add this in.